python/pyspark/errors/exceptions/base.py

```diff
@@ -31,14 +31,12 @@ def __init__(
         error_class: Optional[str] = None,
         message_parameters: Optional[Dict[str, str]] = None,
     ):
-        # `message` vs `error_class` & `message_parameters` are mutually exclusive.
-        assert (message is not None and (error_class is None and message_parameters is None)) or (
-            message is None and (error_class is not None and message_parameters is not None)
-        )
-
-        self.error_reader = ErrorClassesReader()
-
         if message is None:
+            assert (
```

Comment:
we are losing the assert on `message is not None and (error_class is None and message_parameters is None)`?

Reply:
Yes cc @ueshin I am wondering why do we need this assert?

Reply:
It's just moved from the previous commit.
cc @itholic for the original author.

Reply:
This follows the behavior on the JVM side. All errors should have either an error class and message parameters, or just a message.

Reply:
Why can't these errors include all of the following: error messages, error classes, parameters, and SQL states? If the error message field is None, we can render the message from the error class and parameter. This approach allows us to check for error classes in PySpark unit tests as well.

Reply:
> If the error message field is None, we can render the message from the error class and parameter.

Yes, the existing code already does that. We're doing basically:
1. if the `message` is `None`, we render the message from the `error_class` and `message_parameter`.
2. if the `message` exists, we use that message instead of `error_class` and `message_parameter`. In this case, `error_class` and `message_parameter` will never be used, so we block to accept them in force.

Your fix removes the second condition, so that's why @amaliujia concerned about this I guess.

btw, what is the reason for changing this code? Is there a case where we need to receive `message`, `error_class`, and `message_parameter` at the same time? If we cannot avoid it, I agree with this modification.